### PR TITLE
MGDCTRS-603: Duplicate a connector configuration

### PR DESCRIPTION
**Pull request by indraraj**
Closes issue 
https://issues.redhat.com/browse/MGDCTRS-603
https://issues.redhat.com/browse/MGDCTRS-662

### CODE REVIEW

**Comment by indraraj** [comment]
Hey @gashcrumb, @uidoyen is not here this week can you review this, please.

File: src/app/components/CreateServiceAccount/CreateServiceAccount.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ type CreateServiceAccountProps = {
   isOpen: boolean;
   sACreated: boolean;
   handleModalToggle: () => void;
-  serviceAccount: UserProvidedServiceAccount;
+  serviceAccount: any;
```
Comment (gashcrumb) [inline src/app/components/CreateServiceAccount/CreateServiceAccount.tsx:37]: Is it needed to relax the type on this?

**Review by gashcrumb — APPROVED** [review]
Just a couple questions that'd be good to answer, overall LGTM though, thanks!

File: src/utils/shared.ts [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+
 export const mapToObject = (inputMap: Map<string, unknown>): object => {
   const obj = {} as { [key: string]: unknown };
   inputMap.forEach((value, key) => {
     obj[key] = value;
   });
   return obj;
 };
+
+export const clearSecretEmptyValue = (obj: any) => {
```
[the remark this answers is not in the document]
Comment (indraraj) [reply src/utils/shared.ts:11]: so for secret fields backend does not return the value in response but an empty object is a returned and on the UI we show those fields as a disabled input text. so this function is replacing `{}` with an empty string value.

File: src/app/pages/CreateConnectorPage/StepKafkas.tsx [hunk 3]
```diff
@@ -71,6 +72,17 @@ const KafkasGallery: FunctionComponent = () => {
...
 
+  useEffect(() => {
+    if (duplicateMode && response) {
+      if (response?.items?.find((i) => i.id === selectedId)) {
+        onSelect(selectedId!);
+      } else {
+        onDeselect();
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (indraraj) [reply src/app/pages/CreateConnectorPage/StepKafkas.tsx:83]: here we don't want the useEffect to run on selectedId update

File: src/app/components/CreateServiceAccount/CreateServiceAccount.tsx (hunk 1 above)
Comment (indraraj) [reply src/app/components/CreateServiceAccount/CreateServiceAccount.tsx:37]: I will update this

**Comment by indraraj** [comment]
@gashcrumb I am merging this to reduce the conflicts in namespace work. if you see any other issues you can comment them here, We can pick them up in subsequent PR.

File: src/app/components/CreateServiceAccount/CreateServiceAccount.tsx (hunk 1 above)
Comment (indraraj) [reply src/app/components/CreateServiceAccount/CreateServiceAccount.tsx]: Done

**Comment by gashcrumb** [comment]
yep, tested and all good from here, thanks!

File: src/utils/shared.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gashcrumb) [reply src/utils/shared.ts:11]: ok cool, thanks!

**Pull request merged**